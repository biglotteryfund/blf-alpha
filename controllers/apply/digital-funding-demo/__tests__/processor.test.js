/* eslint-env jest */
'use strict';
const nodemailer = require('nodemailer');

// Mock email secret
process.env.DIGITAL_FUND_DEMO_EMAIL = 'digital@example.com';

const processor = require('../processor');
const form = require('../form-model');
const { flattenFormData, stepsWithValues } = require('../../helpers');

const mockFormData = {
    'step-1': {
        name: 'Example Person',
        email: 'example@example.com',
        'organisation-name': 'Test Organisation',
        'about-your-organisation': 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro, inventore.'
    }
};

function cleanMailForSnaphot(info) {
    info.message = info.message
        .toString()
        .replace(/Message-ID: .*\n/, '')
        .replace(/Date: .*\n/, '')
        .replace(/ boundary=.*\n/, '')
        .replace(/----.*\n/gm, '');
    return info;
}

describe('processor', () => {
    it('should create success emails', async () => {
        const mockTransport = nodemailer.createTransport({
            streamTransport: true,
            newline: 'unix',
            buffer: true
        });

        const results = await processor({
            form: form,
            data: flattenFormData(mockFormData),
            stepsWithValues: stepsWithValues(form(1).steps, mockFormData),
            mailTransport: mockTransport
        });

        const [customerEmail] = results.map(cleanMailForSnaphot);
        expect(customerEmail.message).toMatchSnapshot();
    });
});
