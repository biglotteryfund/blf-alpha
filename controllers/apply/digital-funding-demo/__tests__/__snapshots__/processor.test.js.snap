// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processor should create success emails 1`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
Subject: Thank you for getting in touch with the Big Lottery Fund!
MIME-Version: 1.0

Content-Type: text/plain
Content-Transfer-Encoding: quoted-printable

Thank you for getting in touch with us.

You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in
touch to let you know if we can help.

In the meantime, you can find out more about our funding and how we can =
support your community on our website:=20
www.biglotteryfund.org.uk/england=
 [https://www.biglotteryfund.org.uk/england]

Thanks,

The Big Lottery Fund

YOUR INFORMATION
YOUR DETAILS
DATA PROTECTION
The Big Lottery Fund is a =
public body with a duty to distribute National Lottery and other money in =
grants for good causes. We use
the personal data you provide, such as =
contact details for individuals at your organisation, to help you apply for=
 a grant and to
assess your application. We may share your personal data =
with organisations which help us with our grant making activities or
others which have a legitimate interest in our work or have funded your =
grant. We will only share personal data which they need to
carry out their work and subject to appropriate safety measures. Our Data =
Protection and Privacy Notice gives more information
about how we store and use personal data and the lawful basis for this. =
Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection or contact us to =
request a hard copy. The Notice may be updated from time to
time.
Content-Type: text/html
Content-Transfer-Encoding: quoted-printable

<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"margin: auto; =
max-width: 75%; padding: 20px; color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">
        <div class=3D\\"logo\\" =
style=3D\\"display: block; margin-bottom: 20px; text-align: right;\\">
            <img src=3D\\"https://www.biglotteryfund.org.=
uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">Thank you for getting in touch with =
us.</p>

        <p style=3D\\"color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in touch to let you know if we can help.</p>
        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">In the meantime, you can find out =
more about our funding and how we can support your community on our =
website: <a href=3D\\"https://www.biglotteryfund.org.uk/england\\" =
style=3D\\"color: #365cc9;\\">www.biglotteryfund.org.uk/england</a></p>
        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">Thanks,</p>
        <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\"><strong>The Big Lottery Fund</strong></p>

        <h2 class=3D\\"space-above\\" style=3D\\"color: #e20087; font-weight: =
500; margin: 0 0 20px; font-size: 22px; margin-top: 50px;\\">Your =
information</h2>

        <div class=3D\\"summary\\">
           =20
                <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: =
16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your details</h4>
               =20
                   =20
                       =20
                   =20
                       =
=20
                   =20
                       =20
                   =20
               =20
                   =20
                       =20
                   =20
               =20
           =20
        </div>

        <h5 class=3D\\"space-above\\" =
style=3D\\"color: #e20087; font-weight: 500; margin: 0 0 20px; font-size: =
14px; margin-top: 50px;\\">Data protection</h5>
        <small =
style=3D\\"color: #666; display: block; font-size: 11px; margin-bottom: =
10px;\\">The Big Lottery Fund is a public body with a duty to distribute =
National Lottery and other money in grants for good causes. We use the =
personal data you provide, such as contact details for individuals at your =
organisation, to help you apply for a grant and to assess your application.=
</small>
        <small style=3D\\"color: #666; display: block; font-size: =
11px; margin-bottom: 10px;\\">We may share your personal data with =
organisations which help us with our grant making activities or others =
which have a legitimate interest in our work or have funded your grant. We =
will only share personal data which they need to carry out their work and =
subject to appropriate safety measures.</small>
        <small =
style=3D\\"color: #666; display: block; font-size: 11px; margin-bottom: =
10px;\\">Our Data Protection and Privacy Notice gives more information about =
how we store and use personal data and the lawful basis for this. Please =
read the full Notice which is published on our website at www.=
biglotteryfund.org.uk/data-protection or contact us to request a hard copy.=
 The Notice may be updated from time to time.</small>
    </body>
</html>

"
`;

exports[`processor should create success emails 2`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@blf.digital>
To: digitalfund@biglotteryfund.org.uk
Subject: New Digital Funding idea submission from website: undefined
MIME-Version: 1.0

Content-Type: text/plain
Content-Transfer-Encoding: 7bit

A customer has submitted an idea from the web form.

The details they provided are included below:

CUSTOMER INFORMATION
YOUR DETAILS
Content-Type: text/html
Content-Transfer-Encoding: quoted-printable

<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"margin: auto; =
max-width: 75%; padding: 20px; color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">
        <p style=3D\\"color: =
#333; font-family: arial; font-size: 20px; font-weight: 400; line-height: 1=
.35;\\">A customer has submitted an idea from the  web form.</p>
        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">The details they provided are =
included below:</p>

        <h2 class=3D\\"space-above\\" style=3D\\"color: =
#e20087; font-weight: 500; margin: 0 0 20px; font-size: 22px; margin-top: =
50px;\\">Customer information</h2>

        <div class=3D\\"summary\\">
           =20
                <h4 style=3D\\"color: #e20087; margin: 0 0 =
20px; font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: =
3px; font-weight: 700;\\">Your details</h4>
               =20
                   =20
                       =20
                   =20
                       =20
                   =20
                       =
=20
                   =20
               =20
                   =20
                       =20
                   =20
               =20
           =20
        </div>
    </body>
</html>

"
`;
